applications/h264/grpc_h264_endoscopy_tool_tracking/README.md

```diff
@@ -0,0 +1,93 @@
+
+On the client side, the differences are the queues and the gRPC client. In the *Video Input Fragment*, we added the following:
+- **Outgoing Requests** operator (`GrpcClientRequestOp`): It converts the video frames (GXF entities) received from the *Video Read Stream* operator into `EntityRequest` protobuf messages and queues each frame in the *Request Queue*.
+- **gRPC Service & Client** (`EntityClientService` & `EntityClient`): The gRPC Service is responsible for controlling the life cycle of the gRPC client. The client connects to the remote gRPC server and then sends the requests found in the *Request Queue*. When it receives a response, it converts it into a GXF entity and queues it in the *Response Queue*.
+- **Incoming Responses** operator (`GrpcClientResponseOp`): This operator is configured with an `AsynchronousCondition` condition to check the availability of the *Response Queue*. When notified of available responses in the queue, it dequeues each item and emits each to the output port.
+
+![App Cloud](static/app-cloud.png)
+*Details of App Cloud*
+
+The App Cloud (the server) application consists of a gRPC server and a few components for managing Holoscan applications. When the server receives a new remote procedure call in this sample application, it launches a new instance of the Endoscopy Tool Tracking application. This is facilitated by the `ApplicationFactory` used for application registration.
+
+ Under the hood, the Endoscopy Tool Tracking application here inherits a custom base class (`HoloscanGrpcApplication`) which manages the `Request Queue` and the `Response Queue` as well as the `GrpcServerRequestOp` and `GrpcServerResponseOp` operators for receiving requests and serving results, respectively. When the RPC is complete, the instance of the Endoscopy Tool Tracking application is destroyed and ready to serve the subsequent request.
+
+
+## Requirements
+
+This application is configured to use H.264 elementary stream from endoscopy sample data as input.
+
+### Data
+
+[📦️ (NGC) Sample App Data for AI-based Endoscopy Tool Tracking](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/clara-holoscan/resources/holoscan_endoscopy_sample_data)
+
+The data is automatically downloaded when building the application.
+
+## Building and Running gRPC H.264 Endoscopy Tool Tracking Application
+
+* Building and running the application from the top level Holohub directory:
+
+### C++
+
+```bash
+# Start the gRPC Server
+./dev_container build_and_run grpc_h264_endoscopy_tool_tracking --run_args cloud [--language cpp]
+
+# Start the gRPC Client
+./dev_container build_and_run grpc_h264_endoscopy_tool_tracking --run_args edge [--language cpp]
+```
+
+Important: on aarch64, applications also need tegra folder mounted inside the container and
```

Comment:
I have yet to test this on IGX, but for all H. 264 apps, this line in the Dockerfile needs to be commented out. 
```
#ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/aarch64-linux-gnu/tegra/
```
It doesn't hurt to have it set regardless of the system type.

Reply:
Sure, that's fine. I'm not certain whether this will work out of the box on IGX dGPU with that line alone uncommented, but not a major concern if the app isn't currently targeting or tested on IGX dGPU.

Reply:
I tested it on IGX dGPU and uncommenting the line is required to work and yes, it worked.

Reply:
Great! On inspection I see we're explicitly mounting that directory in the `dev_container` script, so no concerns.

https://github.com/nvidia-holoscan/holohub/blob/main/dev_container#L291